src/main/java/com/softserve/bookstore/repositories/UserRepository.java

```diff
@@ -2,123 +2,148 @@
-        PreparedStatement userStatement = connection.prepareStatement(SELECT_LAST_USERS);
-        userStatement.setInt(1, numberOfRecords);
-        ResultSet resultSet = userStatement.executeQuery();
-        List<User> users = UserUtility.getUsersFromResultSet(resultSet);
-        Collections.reverse(users);
+    private  List<User> getUsersFromResultSet(ResultSet resultSet) throws SQLException {
+        List<User> users = new ArrayList<>();
+        while(resultSet.next()){
+            int userId = resultSet.getInt(User.FIELD_USER_ID);
+            String email = resultSet.getString(User.FIELD_EMAIL).trim();
+            String password = resultSet.getString(User.FIELD_PASSWORD).trim();
+            User user = new User(userId, email, password);
+            users.add(user);
+        }
         return users;
     }
 
+    private void addRole(String roleName, User user) throws SQLException {
+        Connection connection = connectionManager.getConnection();
+
+        Map<String, Integer> roleIds = new HashMap<>();
+        roleIds.put(Role.USER.toString(), 1);
+        roleIds.put(Role.ADMIN.toString(), 2);
+
+        int roleId = Optional.ofNullable(roleIds.get(roleName))
+                .orElseThrow(() -> new IllegalArgumentException("Invalid role name."));
+
+        List<Role> roles = new ArrayList<>();
+        roles.add(Role.valueOf(roleName));
+        user.setRoles(roles);
+
+        PreparedStatement roleStatement = connection.prepareStatement(INSERT_USERS_ROLES);
+        roleStatement.setInt(1, user.getUserId());
+        roleStatement.setInt(2, roleId);
+        roleStatement.executeUpdate();
+        Logger.info("User " + user.getEmail() + " has now role " + user.getRoles());
+    }
+
+    private void addOrder(Order order, User user, PreparedStatement orderStatement) throws SQLException {
+        orderStatement.setInt(1, user.getUserId());
+        orderStatement.setDate(2, order.getDate());
+        orderStatement.setString(3, order.getStatus().toString());
+    }
+
     private void addOrdersToUser(User user, int numberOfRecords) throws SQLException, UserNotFoundException {
         Connection connection = connectionManager.getConnection();
         PreparedStatement orderStatement = connection.prepareStatement(INSERT_ORDERS);
 
-        List<User> users =  findLastUsersAdded(numberOfRecords);
+        List<User> users = findLastUsersAdded(numberOfRecords);
         User correspondingUser = users.stream()
                 .filter(u -> u.getEmail().equals(user.getEmail()))
                 .findFirst()
                 .orElseThrow(() -> new UserNotFoundException("User does not exist."));
 
         user.getOrders().forEach(order -> {
             try {
-                orderRepository.addOrder(order, correspondingUser, orderStatement);
+                addOrder(order, correspondingUser, orderStatement);
                 orderStatement.addBatch();
             } catch (SQLException e) {
```

Comment:
please don't forget to add a specific message to exception.
so if you throw it to controller layer, it would contain the reason of exception
in this case exception message would be something like this
```
throw new RuntimeExceptione("Exception during adding order, SQLException - " + e.getMessage)
```